### PR TITLE
add formatted docstrings for python api

### PR DESCRIPTION
**Pull request by axl1313**
This PR adds docstrings for all user facing Python API functions in the format required for autogenerating the API reference on our new docs site.

To view generated API reference:
1. checkout this branch of `cleanlab-studio`
2. run `pip install --editable .`
3. clone `cleanlab-studio-docs` repo
4. run `make install` from inside `cleanlab-studio-docs`
5. run `make update-api-reference`
6. run `make docs`
7. view docs at localhost:3000

### CODE REVIEW

**Review by anishathalye — COMMENTED** [review]
- [ ] Is there a way to hide components of the generated docs? E.g., the global variable `pyspark_exists`?
- [ ] Can we change the title of the rendered docs to say "cleanlab_studio" rather than "cleanlab_studio.studio.studio" (or just render docs for the `cleanlab_studio` module directly)? Our code re-exports `cleanlab_studio.studio.studio.Studio` as `cleanlab_studio.Studio`.
- [ ] nit: can we put a period at the end of the summary sentence / end of each arg's description / end of the return value description? Relatedly, we can add our own documentation style guide [like cleanlab's](https://github.com/cleanlab/cleanlab/blob/master/DEVELOPMENT.md#documentation-style).
- [ ] Can you demonstrate how to (a) hyperlink to another part of the Python reference, and (b) hyperlink to another part of the docs (e.g., we will likely want to link the docs for download_cleanlab_columns to an explanation of all the cleanlab columns)

Other than these pretty minor things, docs are looking great overall!

**Comment by axl1313** [comment]
> Is there a way to hide components of the generated docs? E.g., the global variable pyspark_exists?

We can hide things like `pyspark_exists` by making them private variables (prefixing with _)

> Can we change the title of the rendered docs to say "cleanlab_studio" rather than "cleanlab_studio.studio.studio"

Just updated the docs code to do this

**Comment by axl1313** [comment]
> Can you demonstrate how to (a) hyperlink to another part of the Python reference, and (b) hyperlink to another part of the docs (e.g., we will likely want to link the docs for download_cleanlab_columns to an explanation of all the cleanlab columns)

Added a quick guide for how to do this in `DEVELOPMENT.md` and some examples in the existing docstrings

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -155,4 +155,9 @@ Every time the version number is incremented, these parts of the codebase need t
 
 ## Documentation Style Guide
 
-cleanlab-studio uses [Google style](https://google.github.io/styleguide/pyguide.html#381-docstrings) docstrings ([example](https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html))
+We autogenerate documentation for the `studio` module from docstrings. In order for the generated docs to be formatted correctly, use [Google style](https://google.github.io/styleguide/pyguide.html#381-docstrings) docstrings ([example](https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html))
+
+### Relative Link Formatting
+
+- To link to other functions in the `studio` module within a docstring, use the following format: `[link text](#method-function_name)`
+- To link to another page in Cleanlab Studio docs use `[link text](../../{path within cleanlab-studio-docs/docs})`
```
Comment (anishathalye) [inline DEVELOPMENT.md:163]: Should these be relative links, or would absolute links like`/guide` also work and be more convenient?

File: cleanlab_studio/studio/studio.py [hunk 2]
```diff
@@ -104,7 +104,7 @@ def download_cleanlab_columns(
 
     def apply_corrections(self, cleanset_id: str, dataset: Any, keep_excluded: bool = False) -> Any:
         """
-        Applies corrections from a Cleanlab Studio cleanset to your dataset. Corrections can be made by viewing your project in the Cleanlab Studio webapp.
+        Applies corrections from a Cleanlab Studio cleanset to your dataset. Corrections can be made by viewing your project in the Cleanlab Studio webapp (see [Cleanlab Studio web quickstart](../../guide/quickstart/web#review-the-errors)).
```
Comment (anishathalye) [inline cleanlab_studio/studio/studio.py:107]: It seems that absolute links also work, `/guide/quickstart/web#review-the-errors`, should we use those instead?

File: DEVELOPMENT.md (hunk 1 above)
Comment (axl1313) [reply DEVELOPMENT.md:163]: Ah yeah I think absolute paths from within the docs folder should work

**Comment by axl1313** [comment]
> This still isn't working for me. If it's working for you, we can hop on a huddle and figure out what's going on.
Screenshot 2023-07-18 at 10 55 28 AM
This is what it looks like for me. Not sure if this is what you meant by changing the title, or if you want the module name to also say `cleanlab_studio`

**Comment by anishathalye** [comment]
I see the same thing. Can we get `module cleanlab_studio.studio.studio` to say just `module cleanlab_studio`? Either by "replacing the text", or by actually building docs for the `cleanlab_studio` module. We re-export `cleanlab_studio.studio.studio.Studio` as `cleanlab_studio.Studio`, and that's how we want people to import it. Feels a bit silly to show `cleanlab_studio.studio.studio`.

**Comment by axl1313** [comment]
> I see the same thing. Can we get `module cleanlab_studio.studio.studio` to say just `module cleanlab_studio`? Either by "replacing the text", or by actually building docs for the `cleanlab_studio` module. We re-export `cleanlab_studio.studio.studio.Studio` as `cleanlab_studio.Studio`, and that's how we want people to import it. Feels a bit silly to show `cleanlab_studio.studio.studio`.

This should be fixed now!

**Comment by anishathalye** [comment]
LGTM! You can go ahead and merge.

**Pull request merged**